presto-main/src/main/java/com/facebook/presto/sql/planner/plan/DistinctLimitNode.java

```diff
@@ -43,13 +43,15 @@ public DistinctLimitNode(
             @JsonProperty("source") PlanNode source,
             @JsonProperty("limit") long limit,
             @JsonProperty("partial") boolean partial,
+            @JsonProperty("distinctSymbols") List<Symbol> distinctSymbols,
```

Comment:
distinctSymbols should only indicate which symbols are used to compute whether two rows are distinct from each other. This should not include the hash. If the hash symbol is present, it just allows the execution to reuse the hash and avoid some unnecessary computation.